app/test_helpers.go

```diff
@@ -75,15 +76,15 @@ type SetupOptions struct {
 	ChainID            string
 }
 
-func setup(t *testing.T, withGenesis bool, invCheckPeriod uint) (*App, GenesisState) {
+func setup(t *testing.T, withGenesis bool, invCheckPeriod uint, chainID string) (*App, GenesisState) {
 	db := dbm.NewMemDB()
 	encCdc := MakeEncodingConfig()
 	// set default config if not set by the flow
 	if len(pioconfig.GetProvenanceConfig().FeeDenom) == 0 {
 		pioconfig.SetProvenanceConfig("", 0)
 	}
 
-	app := New(loggerMaker(), db, nil, true, map[int64]bool{}, t.TempDir(), invCheckPeriod, encCdc, simtestutil.EmptyAppOptions{})
+	app := New(loggerMaker(), db, nil, true, map[int64]bool{}, t.TempDir(), invCheckPeriod, encCdc, simtestutil.EmptyAppOptions{}, baseapp.SetChainID(chainID))
```

Comment:
Does this work when `chainID` is `""`? Should we have it use a default instead?

Reply:
It defaults to that when not set.  However, we have some tests that are setting the chain-id  in one area but not the app level.  In v50, on init chain it checks to see if the chainid match.

Test: https://github.com/provenance-io/provenance/blob/fa05e85e513a05dce5c6015152296d96657d0e75/internal/handlers/msg_service_router_test.go#L234

Check: https://github.com/provenance-io/cosmos-sdk/blob/6e4cf880fcdffb7ec38b3f8704880e7bad01da32/baseapp/abci.go#L40

Perhaps, we could just remove the naming of the chain from the tests and revert my other changes?  I was just trying not to change too much on how the tests were setup.